src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -343,16 +344,30 @@ void ClassPathZipEntry::contents_do(void f(const char* name, void* context), voi
 
 DEBUG_ONLY(ClassPathImageEntry* ClassPathImageEntry::_singleton = NULL;)
 
+JImageFile* ClassPathImageEntry::jimage() const {
+  return JImage_File;
+}
+
+JImageFile* ClassPathImageEntry::jimage_non_null() const {
+  assert(ClassLoader::has_jrt_entry(), "must be");
+  assert(jimage() != NULL, "should have been opened by ClassLoader::lookup_vm_options "
+                           "and remained throughout normal JVM lifetime");
+  return jimage();
+}
+
+void ClassPathImageEntry::set_jimage(JImageFile* jimage) {
+  JImage_File = jimage;
+}
```

Comment:
This setter seems overkill, especially when it is only used to clear the field, and there is no setter used when giving the field a real value.